presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -98,12 +98,28 @@ public DwrfMetadataReader(RuntimeStats runtimeStats, OrcReaderOptions readerOpti
     }
 
     @Override
-    public PostScript readPostScript(byte[] data, int offset, int length)
+    public PostScript readPostScript(OrcDataSourceId orcDataSourceId, byte[] data, int offset, int length)
             throws IOException
     {
+        // corrupted or non-ORC files often have some garbage at the end of the file instead of the PostScript length
+        if (length == 0) {
```

Comment:
Looks like the minimal size of a serializes protobuf is 2 bytes, but we will stick to 0.